src/app/hooks.ts

```diff
@@ -0,0 +1,5 @@
+import { useDispatch, useSelector } from "react-redux";
+import { AppDispatch, RootState } from "./store";
+
+export const useAppSelector = useSelector.withTypes<RootState>();
```

Comment:
The `useSelector` hook does not have a `withTypes` method. Instead, you should use `useSelector` with a type argument directly like this: `export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;`.